File: python/2.2 Algoritmos de Ordenamiento/insercion/InsercionTestV2.py
import numpy as np

datos = np.array([8,5,3,9,1])
print ("Arreglo original: " + str(datos))

for recorrido in range(1, datos.size, 1):
	buffer = datos[recorrido]
	indice = recorrido 
	
	while ( (indice > 0) and (datos[indice - 1] > buffer)): 
		datos[indice] = datos[indice - 1] 
		indice -= 1
	
	datos[indice] = buffer

print ("Arreglo ordenado: " + str(datos))

File: c++/2.2 Algoritmos de Ordenamiento/insercion/InsercionTestV2.cpp
#include <iostream>

using namespace std;
int main(int argc, char *argv[]) {
	int datos[] = {8,5,3,9,1};
	
	cout << "Arreglo original: [";
	copy(begin(datos), end(datos), ostream_iterator<int>(cout, ","));
	cout << "\b]" << endl;
	
	int length = sizeof(datos) / sizeof(datos[0]);
	
	for (int recorrido = 1; recorrido < length; recorrido++) {
		int buffer = datos[recorrido];
		int indice = recorrido;
		
		while ( (indice > 0) && (datos[indice - 1] > buffer)) {
			datos[indice] = datos[indice - 1] ;
			indice--;
		}
		datos[indice] = buffer;
	}
	
	cout << "Arreglo ordenado: [";
	copy(begin(datos), end(datos), ostream_iterator<int>(cout, ","));
	cout << "\b]";
	
	return 0;
}

